src/mapdata.cpp

```diff
@@ -1564,19 +1564,9 @@ void activity_data_common::load( const JsonObject &jo )
     optional( jo, was_loaded, "byproducts", byproducts_ );
 
     optional( jo, was_loaded, "prying_data", prying_data_ );
-}
-
-void activity_data_ter::load( const JsonObject &jo )
-{
-    mandatory( jo, was_loaded, "result", result_ );
-    activity_data_common::load( jo );
-    valid_ = true;
-}
 
-void activity_data_furn::load( const JsonObject &jo )
-{
-    optional( jo, was_loaded, "result", result_, furn_str_id::NULL_ID() );
-    activity_data_common::load( jo );
+    // was_loaded==true won't overwrite result_ if it's missing from the json
+    optional( jo, true, "result", result_ );
```

Comment:
Why do you want this? `result_` is initialized to `ter_str_id()` either way.

Reply:
result_ is a std::variant and at this point it will have previously been primed with either a ter_str_id or a furn_str_id, and here we need to NOT reset the furn-primed version to be ter-primed

Reply:
Sorry, I missed that.

Reply:
I agree. It might be warranted if variant was used more than 3 or 4 times, but it is not.